### DISCUSSION TITLE
Sensor to monitor OpExecutionContext for dbt assets

### DISCUSSION THREAD

**Discussion by fredonia88** [dagster-io/dagster#21178] [Q&A]
I have a `@dbt_asset` provisioned with dbt retry logic. Before the asset issues a dbt retry, it triggers a slack alert, but I'd like to expose the slack alert as a sensor I can turn on and off. 

My question is: how do I create a sensor that monitors OpExecutionContext? My thinking is before dbt retry is run, the op can add output metadata or a log. The sensor can monitor the op context for the metadata key, where it will issue a slack alert if the key is found. 

Is this possible?

**Comment by garethbrickman** [comment]
Sensors in Dagster are typically used to monitor external sources or events and then trigger runs based on changes in those sources. They do not directly monitor the execution context of an operation within a run.

If you want to trigger a Slack alert before issuing a dbt retry and have that logic encapsulated within a sensor, you would need to implement custom logic within your dbt asset to emit a specific event or log message that your sensor can detect. However, this would require a custom implementation, as the current documentation does not provide a built-in way to create a sensor that directly monitors `OpExecutionContext` for specific metadata keys or logs.
